src/jsm.c

```diff
@@ -1978,7 +1978,7 @@ js_directGetMsgToJSMsg(const char *stream, natsMsg *msg)
     if ((natsMsg_GetDataLength(msg) == 0)
         && (natsMsgHeader_Get(msg, STATUS_HDR, &val) == NATS_OK))
     {
-        if (strcmp(val, NOT_FOUND_STATUS) == 0)
+        if (strcmp(val, HDR_STATUS_NOT_FOUND_404) == 0)
```

Comment:
All the errors should be communicated to the user (including `missing heartbeats` which should happen after missing hearbeat set duration x2), but none should terminate the proccess.

Reply:
Again this came out of preserving the current APIs and attempting at consistency with them (`js_Fetch` is effectively unchanged by this PR).

Reply:
Discussed offline, no change